bus_sdk/publishers/sns_publisher.py

```diff
@@ -0,0 +1,67 @@
+try:
+    import boto3
+except ImportError:
+    # boto3 not yet available, may happen in initial install of bus_sdk package
+    pass
+
+from bus_sdk.messages import Message
+from bus_sdk.publishers.event_publisher import EventPublisher
+
+
+class SNSPublisher(EventPublisher):
+    name = 'sns'
+
+    def __init__(self, region: str = '', subscriber: str = '', name: str = '',
+                 env: str = '', **overrides) -> None:
+        """
+        Allows publication of `Message` objects to a target AWs SNS Topic
+
+        :param region: str
+        :param subscriber: str
+        :param name: str
+        :param env: str
+        :param overrides: dict: Any boto3.resource override values that you would like to pass
+        """
+        self._arn = self._create_arn(region, subscriber, name, env)
+        self._resource = boto3.resource(self.name, **overrides)
+
+    @property
+    def arn(self):
+        return self._arn
+
+    @staticmethod
+    def _create_arn(region: str, subscriber: str, name: str, env: str) -> str:
+        """
+        :param region: str
+        :param subscriber: str
+        :param name: str
+        :param env: str
+        :return: str
+        """
+        return 'arn:aws:sns:{region}:{subscriber}:{name}--{env}'.format(region=region,
+                                                                        subscriber=subscriber,
+                                                                        name=name,
+                                                                        env=env)
+
+    def publish(self, message: Message) -> dict:
+        """
+        Publishes a JSON representation of `Message` object to the target AWS SNS Topic.
+
+        :param message: :class: `Message`
+        :return: dict:
+
+        example return value:
```

Comment:
Different implementations would return different things, so shouldn't be returning an implementation-specific response (not sure there's anything useful here?).

Reply:
`EventPublisher.publish` says it's a `str` too...

Reply:
OK fair point, at the level of the AbstractBaseClass I will put the expected return type as `Any`.

(yes I just noticed that whilst changing this, it was the original return type, I didn't change to dict)